synapse/storage/state.py

```diff
@@ -356,6 +356,60 @@ def get_member_split(self) -> Tuple["StateFilter", "StateFilter"]:
 
         return member_filter, non_member_filter
 
+    def approx_difference(self, subtrahend: "StateFilter") -> "StateFilter":
+        """
+        Returns a state filter which represents self - subtrahend;
+        if the set of state events given by a state filter F are represented as
+        E(F), then the resultant state filter bears this property:
+
+            E(difference(self, subtrahend)) ⊇ E(self) ∖ E(subtrahend)
+
+        Ideally, this should be the narrowest such state filter, but this
+        function returns an approximation (since, for example, the set of
+        possible state keys is infinite).
```

Comment:
Much as I'm keen for any excuse to use `$\setminus$`... want some bikeshedding?

```suggestion
        Creates an approximate copy of `self` which additionally filters out `subtrahend`.

        This is "approximate" because the resulting filter might accept ..... / reject ...
                
        In symbols, if E(F) denotes the set of state events accepted by a state filter F,
        the new filter we create satisfies
        
            E(approx_difference(self, subtrahend)) ⊇ E(self) ∖ E(subtrahend) .
```

Reply:
what do you think of the new description?